### PR TITLE
Refactor the implementation for Qwen series models using the new DashScope SDK APIs.

### PR DESCRIPTION
**Pull request by jiangsier-xyz**
The design of the Dashscope SDK is evolving towards OpenAI, offering new fields and specifications. Utilize these latest features to refactor the implementation of the Qwen models.

### CODE REVIEW

File: langchain4j-dashscope/pom.xml [hunk 1]
```diff
@@ -30,6 +30,12 @@
             <version>2.2.0</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-collections4</artifactId>
```
Comment (langchain4j) [inline langchain4j-dashscope/pom.xml]: please avoid introducing new dependencies

File: langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenChatModel.java [hunk 2]
```diff
@@ -42,18 +41,14 @@ protected QwenChatModel(String apiKey,
...
-        return Optional.of(result)
-                    .map(GenerationResult::getOutput)
-                    .map(GenerationOutput::getText)
-                    .orElse("Oops, something wrong...[request id: " + result.getRequestId() + "]");
+    protected String sendMessage(String prompt, List<Message> messages) {
+        return QwenHelper.answerFrom(doSendMessage(prompt, messages));
     }
 
-    protected GenerationResult doSendMessage(String prompt) {
+    protected GenerationResult doSendMessage(String prompt, List<Message> messages) {
```
Comment (langchain4j) [inline langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenChatModel.java:51]: why does it has both prompt and a list of messages?

**Review by ghost — CHANGES_REQUESTED** [review]

File: langchain4j-dashscope/pom.xml (hunk 1 above)
Comment (jiangsier-xyz) [reply langchain4j-dashscope/pom.xml]: Done

File: langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenChatModel.java (hunk 2 above)
Comment (jiangsier-xyz) [reply langchain4j-dashscope/src/main/java/dev/langchain4j/model/dashscope/QwenChatModel.java:51]: Qwen uses a unified model to support both chat and language usage scenarios. Dashscope wraps the input of the Qwen model, using the same request but providing two parameters: prompt and messages. 

The prompt appears to transfer raw text to the model. The typical use case for the messages is to store history (however, as shown in the example, the latest query can also be placed at the last index of the list and the model still works fine). 

I'm concerned that Dashscope adds something like "<|user|>:" as a prefix to the query before transferring it to the model, which may not align with what a Language Model expects. Therefore, I use a unified method to send requests for both chat and language scenarios (as it is indeed the same API in the backend). I put the raw text into the 'prompt' field in the language model and the messages into the 'messages' field in the chat model.

**Review by ghost — APPROVED** [review]

**Pull request merged**